spec.md

```diff
@@ -139,13 +152,10 @@ of a JSON serialization, the context attributes and the event data might
 both appear within the same JSON object.
 
 ### Extension Attributes
-CloudEvent producers MAY include additional extension attributes within the
```

Comment:
I think this first sentence (in particular the "MAY") is important.  I'm not sure why this section is being reduced.

Reply:
It is removed per comment on duplication with what's newly added in previous section

Reply:
I agree with removing duplicate text but not the normative statements behind them. I would suggest you try to move your new text into this section since they're both about extensions. Then we don't lose any of the normative language but add the clarifications you're looking for.

Reply:
The normative statements added in the previous generic "context attributes" section applies to extension attributes too since extension attributes are also "context attributes".  Putting it into "extension" section adds implicit restriction/assumption that any future attribute will be an extension, which is not right. We can not predict the future and should allow the possibility of some future attribute to be added as standard attribute rather than extension attribute.

Reply:
@cathyhongzhang there aren't *any* normative statements in your new text though. By "normative" I mean using RFC2119 keywords.

> Putting it into "extension" section adds implicit restriction/assumption that any future attribute will be an extension...

no it doesn't. Any new attribute added to the spec will not be an extension  - it won't be since it's spec-defined. All of the attributes you're talking about in your new text are "extensions" because they are not defined by the spec, which is why it makes more sense to put the text into the "extensions" section. Its location in the spec does not change what identity properties people can add, it just make it more readable to put all extensions text into the same location.

Reply:
I agree with @duglin - and I'm sorry if my review wasn't clear on that. I prefer to merge your proposed text into the extension section.